steward/src/allocation.rs

```diff
@@ -50,114 +35,128 @@ pub async fn get_connections() -> Result<Connections, Error> {
-                    vec![
-                        allocation_precommit(&cosmos_key, &allocation, format_eth_address(cellar_address))
-                            .await?,
-                    ],
-                )
-                .await?;
-            }
-            Err(e) => {
-                error!("Couldn't Send Precommit {:?}", e);
-            }
-        }
+    debug!("building commit and precommit objects");
+    let allocation = to_allocation(
+        tick_range,
+        cellar_address.clone(),
+        eth_gas_price.as_u64(),
+    );
+    let allocation_precommit = &allocation_precommit(validator_address, &allocation, cellar_address).await?;
+
+    debug!("getting cosmos fee");
+    let cosmos_gas_price = config.cosmos.gas_price.as_tuple();
+    let fee = Coin {
+        amount: (cosmos_gas_price.0 as u64).into(),
+        denom: cosmos_gas_price.1,
+    };
 
-        match timeout(Duration::from_secs(100), async {
-            loop {
-                // Checking Pre-commits for validators
-                let res = somm_send::query_allocation_precommits(grpc_client).await;
-                if let Ok(val) = res {
+    match timeout(Duration::from_secs(100), async {
+        debug!("waiting for new vote period to start");
+        loop {
+            let res = somm_send::query_commit_period(grpc_client).await;
+            if let Ok(val) = res {
+                if val.vote_period_start == val.current_height {
                     break val;
                 }
-                sleep(Duration::from_secs(1)).await;
             }
-        })
-        .await
-        {
-            Ok(_) => {
-                // Sending Commits
-                somm_send::send_allocation(
-                    &contact,
-                    delegate_cosmos_address,
-                    cosmos_key,
-                    fee,
-                    vec![allocation],
-                )
-                .await?;
-            }
-            Err(e) => {
-                error!("Couldn't Send Commit {:?}", e);
+            sleep(Duration::from_secs(1)).await;
+        }
+    })
+    .await
+    {
+        Ok(_) => {
+            debug!("sending allocation precommit");
+            let response = somm_send::send_precommit(
+                &contact,
+                cosmos_delegate_address.to_string(),
+                cosmos_delegate_key,
+                fee.clone(),
+                vec![allocation_precommit.to_owned()],
+            )
+            .await?;
+            debug!("precommit response: {:?}", response);
+        }
+        Err(err) => {
+            error!("Couldn't Send Precommit {:?}", err);
+        }
+    }
+
+    match timeout(Duration::from_secs(100), async {
+        debug!("querying precommit");
+        loop {
+            let res = somm_send::query_allocation_precommits(grpc_client).await;
+            match res {
+                Ok(val) => {
+                    if val.len() > 0 && val.contains(allocation_precommit) {
```

Comment:
You should only really need the `contains` check here, it should return very quickly if the length is zero.

Reply:
I recall adding it because of a confusing panic. I'll try it again.

Reply:
fixed!